test/integration/index.js

```diff
@@ -134,7 +134,7 @@ before(async function() {
 	await exec("node ../../packages/master config set master.auth_secret TestSecretDoNotUse");
 	await exec("node ../../packages/master config set master.http_port 8880");
 	await exec("node ../../packages/master config set master.https_port 4443");
-	await exec("node ../../packages/master config set master.tls_bits 1024");
+	await exec("node ../../packages/master config set master.tls_bits 2048");
```

Comment:
Is there any reasoning behind this?

Reply:
in my local server, run build script, use TLS 1024bit key and after that i got some error
```
node ../../packages/ctl slave create-config --id 4 --name slave --generate-token
master: [info] Listening for HTTP on port 8880
master: [fatal] 
master: +---------------------------------------------------------------+
master: | Unexpected error occured while starting master, please report |
master: | it to https://github.com/clusterio/factorioClusterio/issues   |
master: +---------------------------------------------------------------+
master: Error: error:140AB18F:SSL routines:SSL_CTX_use_certificate:ee key too small
master:     at Object.createSecureContext (_tls_common.js:129:17)
master:     at Server.setSecureContext (_tls_wrap.js:1328:27)
master:     at Server (_tls_wrap.js:1186:8)
master:     at new Server (https.js:70:14)
master:     at Object.createServer (https.js:94:10)
master:     at startServer (/root/factorioClusterio/packages/master/master.js:193:738)
```
build script
```bash
#!/bin/bash
npx lerna bootstrap --no-ci
node packages/lib/build_mod --source-dir packages/slave/lua/clusterio_lib --output-dir temp/test/sharedMods
proxychains -f /etc/proxychains.conf curl -o temp/test/sharedMods/subspace_storage_1.99.8.zip -L https://github.com/clusterio/factorioClusterioMod/releases/download/1.99.6/subspace_storage_1.99.8.zip
npx lerna run build
npm run-script ci-cover
codecov -f coverage/lcov.info
npm run-script lint
```

Reply:
I don't like this as a solution, 2048 bit keys took a really long time to generate last I checked. After a lot of digging I've found that setting the OPENSSL_CONF environment variable to the absolute path to a .cnf file with the following content
```
openssl_conf = openssl_init

[openssl_init]
ssl_conf = ssl_sect

[ssl_sect]
system_default = system_default_sect

[system_default_sect]
CipherString = DEFAULT:@SECLEVEL=0
``` 
should allow certificates with 512 bit RSA keys to be used. Either way this change doesn't belong in this PR.

Reply:
in my VM Debian 10, the ssl default conf is `DEFAULT@SECLEVEL=2`, i think change `openssl.cnf` the system configuration is not a good solution, Because modifying the ssl configuration may affect some other aspects of the system
```ini
[system_default_sect]
MinProtocol = TLSv1.2
CipherString = DEFAULT@SECLEVEL=2
```
i write ssl generate script to test need time，the tested virtual machine configuration is as follows, use SATA SSD hard driver
```bash
➜  cat /proc/cpuinfo
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 63
model name      : Intel(R) Xeon(R) CPU E5-2678 v3 @ 2.50GHz
stepping        : 2
microcode       : 0x43
cpu MHz         : 2499.998
cache size      : 30720 KB
physical id     : 0
siblings        : 1
core id         : 0
cpu cores       : 1
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 15
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon nopl xtopology tsc_reliable nonstop_tsc cpuid pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm cpuid_fault invpcid_single pti ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 avx2 smep bmi2 invpcid xsaveopt arat md_clear flush_l1d arch_capabilities
bugs            : cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit
bogomips        : 4999.99
clflush size    : 64
cache_alignment : 64
address sizes   : 43 bits physical, 48 bits virtual
power management:

➜  uname -a
Linux debian 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1+deb10u1 (2020-04-27) x86_64 GNU/Linux

➜  free -m
              total        used        free      shared  buff/cache   available
Mem:            986          83         420           4         482         753
Swap:          1021           0        1021

➜  df -h
文件系统        容量  已用  可用 已用% 挂载点
udev            479M     0  479M    0% /dev
tmpfs            99M  4.3M   95M    5% /run
/dev/sda1        39G  1.7G   35G    5% /
tmpfs           494M     0  494M    0% /dev/shm
tmpfs           5.0M     0  5.0M    0% /run/lock
tmpfs           494M     0  494M    0% /sys/fs/cgroup
tmpfs            99M     0   99M    0% /run/user/0

➜  dd if=/dev/urandom of=/root/1g.bin bs=1M count=1024
记录了1024+0 的读入
记录了1024+0 的写出
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 5.67209 s, 189 MB/s

➜  dd if=/root/1g.bin of=/dev/null
记录了2097152+0 的读入
记录了2097152+0 的写出
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 3.74378 s, 287 MB/s

➜  node -v
v12.20.0
```
test script， generate_ssl_cert is `packages/master/src/generate_ssl_cert.js`:
```javascript
const generateSslCert = require("./generate_ssl_cert");

let tls_cert = "./cert.crt";
let tls_key = "./cert.key";

generateSslCert({
	bits: 2048,
	sslCertPath: tls_cert,
	sslPrivKeyPath: tls_key,
	doLogging: true,
});
```
this script output info is 
```bash
➜  time node index.js
[winston] Attempt to write logs with no transports {"message":"Generating SSL certificate with 2048 bits at ./cert.crt","level":"info"}
[winston] Attempt to write logs with no transports {"message":"Generated 2048bit keypair in: 135.286294 ms","level":"info"}
[winston] Attempt to write logs with no transports {"message":"Generated certificate in: 201.285105 ms","level":"info"}
node index.js  0.48s user 0.04s system 99% cpu 0.519 total
```
I think it takes about 0.4 seconds to generate the certificate is not very slow, and only need to generate the certificate once

Reply:
How weird. When I changed the keysize down from 2048 it made the build close a minute shorter and now it doesn't? What changed hmm?

Note that I didn't mention touching the system's openssl.cnf file. You would place that into a separate file and run node with OPENSSL_CONFIG=/path/to/my.cnf

Reply:
i think is possible that the computing resources provided by the container used for construction are limiting the CPU usage. So it takes more time when the key bit is 2048

Reply:
This change will conflict with the removal of the automatic creation of a TLS certificate in #352.